File: Backend/Application/DTOs/Proposta/PropostaDTO.cs
using Backend.Application.DTOs.ProjetoFreelancer;

namespace Backend.Application.DTOs.Proposta
{
    public class PropostaDTO
    {
        public int Id { get; set; }
        public decimal ValorPorHora { get; set; }
        public DateTime PrazoEntrega { get; set; }
        public decimal ValorTotal { get; set; }
        public string Mensagem { get; set; } = string.Empty;
        public string ItensPropostos { get; set; } = string.Empty;
        public bool TaxaSistemaAdicionadaAoTotal { get; set; }
        public int FreelancerId { get; set; }
        public int ServicoId { get; set; }
        public ICollection<ProjetoFreelancerDTO> ProjetosDestacados { get; set; } = [];
    }
}

File: frontend/src/features/freelancers/dtos/freelancer/PropostaDTO.ts
export interface CriarPropostaDTO {
  valorPorHora: number;
  prazoEntrega: string;
  valorTotal: number;
  mensagem: string;
  itensPropostos: string;
  taxaSistemaAdicionadaAoTotal: boolean;
  servicoId: number;
  projetosDestacados: ProjetosDestacadosDTO[];
}

export interface PropostaDTO {
  id: number;
  valorPorHora: number;
  prazoEntrega: Date;
  valorTotal: number;
  mensagem: string;
  itensPropostos: string;
  taxaSistemaAdicionadaAoTotal: boolean;
  servicoId: number;
  freelancerId: number;
  projetosDestacados: ProjetosDestacadosDTO[];
}

interface ProjetosDestacadosDTO {
  id: number;
  nome: string;
  descricao: string;
  imagemUrl: string;
  link: string;
}
